src/systems/rendering.h

```diff
@@ -67,8 +67,16 @@ class RenderSystem
     };
     static std::vector<RenderHandler> render_handlers;
 };
+
 template<typename COMPONENT, bool TRANSPARENT> Render3DInterface<COMPONENT, TRANSPARENT>::Render3DInterface() { RenderSystem::add3DHandler(this); }
 
+// FIX: This is obviously a very inelegant way to share behavior
+glm::mat4 calculateModelMatrix(glm::vec2 position, float rotation, MeshRenderComponent& mrc);
+glm::mat4 calculateModelMatrix(glm::vec2 position, float rotation, MeshRenderComponent& mrc, float scale);
+ShaderRegistry::ScopedShader lookUpShader(MeshRenderComponent& mrc);
+void activateAndBindMeshTextures(MeshRenderComponent& mrc);
+void drawMesh(MeshRenderComponent& mrc, ShaderRegistry::ScopedShader& shader);
```

Comment:
I think these being in a separate file.  Some kind of render utils. The hope is that by extracting these bits of behavior, we can re-use it in `rotatingModelView.cpp:draw` and `render.cpp:MeshRnderComponent::render3D` without forcing `render3D` to adapt to `draw`, but also to allow both to benefit from improvements / changes to render logic without having to separately track down and update them.

If it's simpler or more cost efficient just to maintain two separate implementations, I don't mind ripping this out.